### ISSUE TITLE
[CN] Crash on constant string

### ISSUE THREAD

**Issue by septract** [rems-project/cerberus#240]
CN version: e62d18a

Crashing input: 
```
void a() { ""; }
```

Result: 
```
% cn string_crash_1.c 
string_crash_1.c:1:12: error: CreateReadOnly: not a constant ctype: Ivalignof('char[1]')
void a() { ""; }
           ^~ 
cn: internal error, uncaught exception:
    Dune__exe__Core_to_mucore.ConversionFailed
    Raised at Dune__exe__Core_to_mucore.assert_error in file "backend/cn/core_to_mucore.ml", line 87, characters 7-29
    Called from Dune__exe__Core_to_mucore.n_action in file "backend/cn/core_to_mucore.ml", line 513, characters 18-86
    Called from Dune__exe__Core_to_mucore.n_paction in file "backend/cn/core_to_mucore.ml", line 581, characters 18-32
    Called from Dune__exe__Core_to_mucore.n_expr in file "backend/cn/core_to_mucore.ml" (inlined), line 695, characters 18-33
    Called from Dune__exe__Core_to_mucore.n_expr in file "backend/cn/core_to_mucore.ml", line 704, characters 29-49
    Called from Dune__exe__Core_to_mucore.normalise_globs in file "backend/cn/core_to_mucore.ml", line 1386, characters 7-214
    Called from Dune__exe__Core_to_mucore.normalise_globs_list.(fun) in file "backend/cn/core_to_mucore.ml", line 1401, characters 16-41
    Called from Dune__exe__Effectful.Make.ListM.mapM in file "backend/cn/effectful.ml", line 26, characters 18-21
    Called from Dune__exe__Core_to_mucore.normalise_file in file "backend/cn/core_to_mucore.ml", line 1503, characters 15-53
    Called from Dune__exe__Main.main in file "backend/cn/main.ml", line 179, characters 22-81
    Re-raised at Dune__exe__Main.main in file "backend/cn/main.ml", line 223, characters 8-73
    Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 24, characters 19-24
    Called from Cmdliner_eval.run_parser in file "cmdliner_eval.ml", line 34, characters 37-44
```

**Comment by kmemarian** [comment]
This particular crash is now fixed by 08919dec61aa3cde1226172dae3a38a8565d5960. But Christopher tells me that string litterals are not really supported by CN at moment (in particular the fact that they are readonly).

**Comment by septract** [comment]
Makes sense. We should probably emit an error or warning at least when they occur in programs.
